src/main.controller.js

```diff
@@ -156,33 +156,37 @@ plugin.controller('wgnConfigCtrl', ['$scope', '$q', '$routeParams', 'znData', 'z
 
 			removeHiddenValues();
 
-			return doSaveConfig($scope.editing.config).then(function () {
-				if ($scope.settings.toggle && !$scope.editing.config.enabled && !('$id' in $scope.editing.config)) {
-					znModal({
-						title: '',
-						template: '<p>Your new configuration was saved!</p><p>It must be manually enabled to be active.</p>',
-						classes: 'config-enable-message',
-						closeButton: false,
-						btns: {
-							'OK': {
-								primary: true
+			return doRunHook('beforeSave', $scope.editing.config).finally(function () {
+				return doSaveConfig($scope.editing.config).then(function () {
+					if ($scope.settings.toggle && !$scope.editing.config.enabled && !('$id' in $scope.editing.config)) {
+						znModal({
+							title: '',
+							template: '<p>Your new configuration was saved!</p><p>It must be manually enabled to be active.</p>',
+							classes: 'config-enable-message',
+							closeButton: false,
+							btns: {
+								'OK': {
+									primary: true
+								}
 							}
-						}
-					});
-				}
+						});
+					}
 
-				return doRunHook('save', $scope.editing.config).finally(function () {
+					return doRunHook('save', $scope.editing.config).finally(function () {
 
-					if ($scope.settings.multi) {
-						doDiscardChanges();
-					} else {
-						doResetTab();
-						$scope.editing.form.$setPristine();
-					}
+						if ($scope.settings.multi) {
+							doDiscardChanges();
+						} else {
+							doResetTab();
+							$scope.editing.form.$setPristine();
+						}
 
-					znMessage('Configuration saved!', 'saved');
-					$scope.saving = false;
+						znMessage('Configuration saved!', 'saved');
+						$scope.saving = false;
+					});
 				});
+			}).catch(function () {
+				znMessage('There was an error saving the configuration!', 'error');
```

Comment:
@wizehive-chrismanjoine if we gunna stick with the finally approach, then I think this message is misleading ... because the config gets saved either way. maybe just remove this.

Reply:
Also since the main use case of a beforeSave is to just add/modify the firebase data before it gets saved, case it doesnt really matter if the config save happens in a then or finally callback. So I'm fine with going with finally until we have a use case in the wild for something that requires the firebase save to not happen if the beforeSave is rejected

Reply:
I removed the catch.